### PR TITLE
Adding support for multiple aggregate groups per check

### LINKED ISSUE #1342: Multiple aggregate groups per check
I was thinking Aggregates 2.0 was going to allow for this, but looking at the documentation "aggregate" only accepts a string. Is there a way to have the same check show up in multiple aggregates? If not any plans on supporting this?

### PR DESCRIPTION
**Pull request by cwjohnston**
Implements feature requested in #1342. Closes #1342.

### CODE REVIEW

File: lib/sensu/server/process.rb [hunk 1]
```diff
@@ -275,26 +275,38 @@ def event_bridges(event)
...
-          :aggregate => aggregate,
-          :client => client,
-          :check => check
-        })
-        aggregate_member = "#{client[:name]}:#{check[:name]}"
-        @redis.sadd("aggregates:#{aggregate}", aggregate_member) do
-          @redis.sadd("aggregates", aggregate)
+        check_aggregate = (check[:aggregate].is_a?(String) ? check[:aggregate] : check[:name])
+        aggregate_list = Array(check[:aggregates] || check_aggregate)
+
```
Comment (portertech) [inline lib/sensu/server/process.rb]: Extra newline?

**Comment by portertech** [comment]
We will need to update sensu-settings to validate check "aggregates" (is array of strings etc.).

**Comment by runningman84** [comment]
please also update the chef cookbook!

**Comment by cwjohnston** [comment]
@runningman84 will you please [open an issue against sensu-chef](https://github.com/sensu/sensu-chef/issues/new) to track that request? Thanks!

**Comment by portertech** [comment]
@cwjohnston need to updated `process_check_result()`, e.g. `if check[:aggregate] || check[:aggregates]`

File: spec/server/process_spec.rb [hunk 2]
```diff
@@ -159,6 +159,37 @@
     end
   end
 
+  it "can aggregate check results across multiple named aggregates" do
+    async_wrapper do
+
```
Comment (portertech) [inline spec/server/process_spec.rb]: You love extra newlines :P

**Comment by RyanJarv** [comment]
Awesome, thanks guy's! Will be making good use of this soon.

**Pull request merged**